mqtt_client/src/MqttClient.ros2.cpp

```diff
@@ -59,6 +61,109 @@ const std::string MqttClient::kRosMsgTypeMqttTopicPrefix =
+  std::string bool_str = mqtt_msg->to_string ();
+  std::transform(str_msg.cbegin(), str_msg.cend(), bool_str.begin(),
+                 ::tolower);
+  if (bool_str == "true" || bool_str == "1") return true;
+  if (bool_str == "false" || bool_str == "0") return false;
+
+  throw std::invalid_argument ("unable to decode string");
+}
+
+bool fixedMqtt2PrimitiveRos(mqtt::const_message_ptr mqtt_msg,
+                            const std::string& msg_type,
+                            rclcpp::SerializedMessage &serialized_msg) {
+  try {
+    if (msg_type == "std_msgs/msg/String") {
+      std_msgs::msg::String msg;
+      msg.data = mqtt_msg->to_string();
+
+      serializeRosMessage(msg, serialized_msg);
+      return true;
+    } else if (msg_type == "std_msgs/msg/Bool") {
+      std_msgs::msg::Bool msg;
+      msg.data = mqtt2bool(mqtt_msg);
+
+      serializeRosMessage(msg, serialized_msg);
+      return true;
+    } else if (msg_type == "std_msgs/msg/Char") {
+      std_msgs::msg::Char msg;
+      msg.data = mqtt2int<int8_t>(mqtt_msg);
+
+      serializeRosMessage(msg, serialized_msg);
+      return true;
+    } else if (msg_type == "std_msgs/msg/UInt8") {
+      std_msgs::msg::UInt8 msg;
+      msg.data = mqtt2int<uint8_t>(mqtt_msg);
+
+      serializeRosMessage(msg, serialized_msg);
+      return true;
+    } else if (msg_type == "std_msgs/msg/UInt16") {
+      std_msgs::msg::UInt16 msg;
+      msg.data = mqtt2int<uint16_t>(mqtt_msg);
+
+      serializeRosMessage(msg, serialized_msg);
+      return true;
+    } else if (msg_type == "std_msgs/msg/UInt32") {
+      std_msgs::msg::UInt32 msg;
+      msg.data = mqtt2int<uint32_t>(mqtt_msg);
+
+      serializeRosMessage(msg, serialized_msg);
+      return true;
+    } else if (msg_type == "std_msgs/msg/UInt64") {
+      std_msgs::msg::UInt64 msg;
+      msg.data = mqtt2int<uint64_t>(mqtt_msg);
+
+      serializeRosMessage(msg, serialized_msg);
+      return true;
+    } else if (msg_type == "std_msgs/msg/Int8") {
+      std_msgs::msg::Int8 msg;
+      msg.data = mqtt2int<int8_t>(mqtt_msg);
+
+      serializeRosMessage(msg, serialized_msg);
+      return true;
+    } else if (msg_type == "std_msgs/msg/Int16") {
+      std_msgs::msg::Int16 msg;
+      msg.data = mqtt2int<int16_t>(mqtt_msg);
+
+      serializeRosMessage(msg, serialized_msg);
+      return true;
+    } else if (msg_type == "std_msgs/msg/Int32") {
+      std_msgs::msg::Int32 msg;
+      msg.data = mqtt2int<int32_t>(mqtt_msg);
+
+      serializeRosMessage(msg, serialized_msg);
+      return true;
+    } else if (msg_type == "std_msgs/msg/Int64") {
+      std_msgs::msg::Int32 msg;
+      msg.data = mqtt2int<int32_t>(mqtt_msg);
+
+      serializeRosMessage(msg, serialized_msg);
+      return true;
+    } else if (msg_type == "std_msgs/msg/Float32") {
+      /// TODO
+    } else if (msg_type == "std_msgs/msg/Float64") {
+      /// TODO
+    }
+  } catch (const std::exception &) {
+  }

```

Comment:
Shouldn't we at least print an error message here? I guess you could also `return false` in here and at the end of the function `return true` by default to save a few lines of code.